File: elementary_computer_science/VTT_THPT/sum_subsegment.py
n, k = map(int, input().split())
a = list(map(int, input().split()))
lmin = n # minimum length
pos = 0
for i in range(n):
	s = 0
	for j in range(i, n):
		s += a[j];
		if (s == k and lmin > j - i + 1):
			lmin = j - i + 1
			pos = i + 1
			break
if (pos == 0): print(0)
else: print(pos, lmin)

File: elementary_computer_science/VTT_THPT/sum_subsegment.cpp
#include <iostream>
#define N 2000
using namespace std;
int n, k, a[N];
int main() {
	cin >> n >> k;
	for (int i = 1; i <= n; ++i) cin >> a[i];
	int lmin = n, pos = 0;
	for (int i = 1; i <= n; ++i) {
		int s = 0;
		for (int j = i; j <= n; ++j) {
			s += a[j];
			if (s == k && lmin > j - i + 1) {
				lmin = j - i + 1;
				pos = i;
			}
		}
	}
	if (pos == 0) cout << 0;
	else cout << pos << " " << lmin;
}